Config to Class

alentoghostflame opened the pull request:
Rather than dealing with a dictionary that IDE's can't understand the contents of, and thus can't autocomplete or warn you when using the wrong types, everything has been moved into a class that IDE's can understand and properly see the variables of.

It may be slightly overkill for this, but:
1: If no config file is found, it will initialize default values and create a new one.
2: If new variables are added to the config class(es) it will automatically use and, if the config is saved, save them to disk.

luxaritas reviewed (changes requested):
Two big things here:
- `conf_manage.py` is a bit of an awkward name - what about `config.py`?
- I've been thinking a bit about the whole situation of defaults and saving/loading. Would it be possible to adjust this so that we nix the config.default.yml, but instead when we serialize the default config to YAML we include comments with descriptions of each field? Even better would be if we could handle outdated configs - you could imagine automatically serializing missing fields

alentoghostflame commented:
* Changed file name.
* Moved into util module.
* Added generated-comments support.
* Switched PyYAML dependency to Ruamel.YAML as discussed.
I can't tell if the method for adding comments is easy and/or jank.

luxaritas commented:
Thanks!
